trillian/migrillian/core/controller.go

```diff
@@ -85,17 +126,21 @@ func (c *Controller) Run(ctx context.Context) error {
 	}
 
 	var wg sync.WaitGroup
+	bufferSize := c.opts.Submitters * c.opts.BatchesPerSubmitter
+	c.batches = make(chan scanner.EntryBatch, bufferSize)
+	defer func() {
+		close(c.batches)
+		wg.Wait()
+	}()
```

Comment:
It is related because it creates batches for every mastership run instead of doing it once. We can get closing-already-closed-channel situation otherwise. But I think I can do this in a separate PR before this one.

Reply:
OK, stet

Reply:
Created #271 to isolate this.